src/pykeen/nn/modules.py

```diff
@@ -1413,6 +1414,65 @@ def _prepare_hrt_for_functional(
         return dict(h=h, r=r, c_r=c_r, t=t)
 
 
+class BoxEInteraction(
+    TranslationalInteraction[
+        Tuple[FloatTensor, FloatTensor],
+        Tuple[FloatTensor, FloatTensor, FloatTensor, FloatTensor, FloatTensor, FloatTensor],
+        Tuple[FloatTensor, FloatTensor],
+    ]
+):
+    """An implementation of the BoxE interaction from [abboud2020]_."""
+
+    func = pkf.boxe_interaction
+
+    relation_shape = ("d", "d", "s", "d", "d", "s")  # Boxes are 2xd (size) each, x 2 sets of boxes: head and tail
+    entity_shape = ("d", "d")  # Base position and bump
+
+    def __init__(self, tanh_map: bool = True, p: int = 2, power_norm: bool = False):
+        r"""
+        Instantiate the interaction module.
+
+        :param tanh_map:
+            Should the hyperbolic tangent be applied to all representations prior to model scoring?
+        :param p:
+            the order of the norm
+        :param power_norm:
+            whether to use the p-th power of the norm instead
+        """
+        super().__init__(p=p, power_norm=power_norm)
+        self.tanh_map = tanh_map
+
+    def _prepare_for_functional(
```

Comment:
got it. we are overwriting the wrong function!

This should be `_prepare_hrt_for_functional`.

Reply:
I am not sure whether we would detect this in our tests - is there a way how to enforce that subclasses implement certain methods? (is `@abstractmethod` sufficient, or is this just some annotation for linters?)